pkgs/applications/misc/xiphos/default.nix

```diff
@@ -5,40 +5,59 @@
 , sword, clucene_core, biblesync
 , gnome-doc-utils
 , libgsf, gconf
-, gtkhtml, libglade, scrollkeeper
-, webkitgtk
+, libglade, scrollkeeper
+, webkitgtk, libsoup
 , dbus-glib, enchant, isocodes, libuuid, icu
+, cmake, itstool, desktop-file-utils, appstream-glib, yelp-tools
+, zip, minizip, pcre, libselinux, xorg, libsepol, libthai, libdatrie
+, libxkbcommon, libepoxy, at-spi2-core, libsysprof-capture, sqlite
+, libpsl, brotli, gtk2, glib, dbus, gtkhtml
 , wrapGAppsHook
-, wafHook
 }:
 
 stdenv.mkDerivation rec {
   pname = "xiphos";
-  version = "4.1.0";
+  version = "4.2.1";
 
   src = fetchFromGitHub {
     owner = "crosswire";
     repo = "xiphos";
     rev = version;
-    sha256 = "14il9k4i58qbc78hcadw3gqy21sb9q661d75vlj6fwpczbzj7x1a";
+    hash = "sha256-H5Q+azE2t3fgu77C9DxrkeUCJ7iJz3Cc91Ln4dqLvD8=";
   };
 
-  nativeBuildInputs = [ pkg-config wrapGAppsHook wafHook ];
-  buildInputs = [ python intltool docbook2x docbook_xml_dtd_412 libxslt
-                  sword clucene_core biblesync gnome-doc-utils libgsf gconf gtkhtml
-                  libglade scrollkeeper webkitgtk dbus-glib enchant isocodes libuuid icu ];
+  nativeBuildInputs = [ pkg-config  wrapGAppsHook
+    appstream-glib # for appstream-util
+    desktop-file-utils # for desktop-file-validate
+    yelp-tools # for yelp-build
+  ];
 
-  prePatch = ''
-    patchShebangs .;
-  '';
+  buildInputs = [
+    python intltool docbook2x docbook_xml_dtd_412 libxslt cmake itstool
+    sword clucene_core biblesync gnome-doc-utils libgsf gconf
+    libglade scrollkeeper webkitgtk dbus-glib enchant isocodes libuuid icu
+    zip minizip pcre xorg.libXdmcp libselinux libsepol libthai libdatrie
+    libxkbcommon libepoxy at-spi2-core libsysprof-capture xorg.libXtst
+    sqlite libpsl brotli gtk2 glib dbus dbus-glib libsoup gtkhtml
+  ];
+
+  cmakeFlags = [
+    "-DDBUS=OFF"
+    "-DGTKHTML=ON"
+  ];
 
   preConfigure =  ''
+    # Installer won't continue without the version saved locally.
```

Comment:
When you say "installer", you are talking about what? The "make install"? Or the build script as a whole?

Reply:
I updated the comment, it was the build script that would not continue